torch_frame/data/dataset.py

```diff
@@ -278,7 +309,10 @@ def __call__(
             y = self._get_mapper(self.target_col).forward(
                 df[self.target_col], device=device)
 
-        return TensorFrame(feat_dict, self.col_names_dict, y)
+        tf = TensorFrame(feat_dict, self.col_names_dict, y)
+        self._update_col_stats(tf)
```

Comment:
It's weird to update col_stats here. It means we update col_stats in every df to tensor frame conversion.